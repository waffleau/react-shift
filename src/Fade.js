import React, {PropTypes, PureComponent} from 'react'

import Transition from './Transition'

export class Fade extends PureComponent {
  static defaultProps = {
    minOpacity: 0,
    maxOpacity: 1
  }

  static propTypes = {
    ...Transition.baseProps,
    maxOpacity: PropTypes.number,
    minOpacity: PropTypes.number

  }

  getFromValue() {
    return {
      opacity: this.props.minOpacity
    }
  }

  getToValue() {
    return {
      opacity: this.props.maxOpacity
    }
  }

  render() {
    const {children} = this.props
    return (
      <Transition
        {...this.props}
        fromValue={this.getFromValue()}
        toValue={this.getToValue()}>
        {children}
      </Transition>
    )
  }
}

export default Fade
