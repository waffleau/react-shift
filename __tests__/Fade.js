import React from 'react'
import renderer from 'react-test-renderer'

import Fade from '../src/Fade'

test('has opacity 0 when inactive', () => {
  const tree = renderer.create(
    <Fade active={false}>
      {'Children'}
    </Fade>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('has opacity 1 when active', () => {
  const tree = renderer.create(
    <Fade active>
      {'Children'}
    </Fade>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a minimum opacity', () => {
  const tree = renderer.create(
    <Fade
      active={false}
      minOpacity={0.1}>
      {'Children'}
    </Fade>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a maximum opacity', () => {
  const tree = renderer.create(
    <Fade
      active
      maxOpacity={0.9}>
      {'Children'}
    </Fade>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
