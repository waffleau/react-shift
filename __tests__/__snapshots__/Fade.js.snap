exports[`test can specify a maximum opacity 1`] = `
<div
  style={
    Object {
      "opacity": 0.9,
      "transition": "opacity 600ms linear 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can specify a minimum opacity 1`] = `
<div
  style={
    Object {
      "opacity": 0.1,
      "transition": "opacity 600ms linear 0ms",
    }
  }>
  Children
</div>
`;

exports[`test has opacity 0 when inactive 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 600ms linear 0ms",
    }
  }>
  Children
</div>
`;

exports[`test has opacity 1 when active 1`] = `
<div
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 600ms linear 0ms",
    }
  }>
  Children
</div>
`;
